File: src/main/java/com/github/tomasmilata/intelliroutes/RoutesFileType.java
package com.github.tomasmilata.intelliroutes;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.*;

import javax.swing.*;

public class RoutesFileType extends LanguageFileType {
  public static final RoutesFileType INSTANCE = new RoutesFileType();

  private RoutesFileType() {
    super(RoutesLanguage.INSTANCE);
  }

  @NotNull
  @Override
  public String getName() {
    return "Routes file";
  }

  @NotNull
  @Override
  public String getDescription() {
    return "Play framework routes file";
  }

  @NotNull
  @Override
  public String getDefaultExtension() {
    return "simple";
  }

  @Nullable
  @Override
  public Icon getIcon() {
    return RoutesIcons.FILE;
  }
}

File: src/main/java/com/github/tomasmilata/intelliroutes/RoutesFileType.kt
package com.github.tomasmilata.intelliroutes

import com.intellij.openapi.fileTypes.LanguageFileType
import org.jetbrains.annotations.*

import javax.swing.*

class RoutesFileType private constructor() : LanguageFileType(RoutesLanguage.INSTANCE) {

    override fun getName(): String {
        return "Routes file"
    }

    override fun getDescription(): String {
        return "Play framework routes file"
    }

    override fun getDefaultExtension(): String {
        return "routes"
    }

    override fun getIcon(): Icon? {
        return RoutesIcons.FILE
    }

    companion object {
        val INSTANCE = RoutesFileType()
    }
}